lib/code_corps/stripe_service/stripe_connect_subscription.ex

```diff
@@ -2,89 +2,79 @@ defmodule CodeCorps.StripeService.StripeConnectSubscriptionService do
   import Ecto.Query
 
   alias CodeCorps.{
-    Organization, Project, Repo, StripeConnectAccount,StripeConnectCard,
-    StripeConnectCustomer, StripeConnectPlan, StripeConnectSubscription,
-    StripePlatformCard, StripePlatformCustomer, User
-  }
-  alias CodeCorps.Services.ProjectService
-  alias CodeCorps.StripeService.{
-    StripeConnectCardService, StripeConnectCustomerService
+    Project, Repo, StripeConnectCard, StripeConnectCustomer, StripeConnectAccount,
+    StripeConnectPlan, StripeConnectSubscription, User
   }
+  alias CodeCorps.Services.{DonationGoalsService, ProjectService}
+  alias CodeCorps.StripeService.{StripeConnectCardService, StripeConnectCustomerService}
   alias CodeCorps.StripeService.Adapters.StripeConnectSubscriptionAdapter
+  alias CodeCorps.StripeService.Validators.{ProjectSubscribable, UserCanSubscribe}
 
   @api Application.get_env(:code_corps, :stripe)
 
-  def find_or_create(%{"project_id" => project_id, "quantity" => quantity, "user_id" => user_id} = attributes) do
-    with %Project{
-           stripe_connect_plan: %StripeConnectPlan{} = plan,
-           organization: %Organization{
-             stripe_connect_account: %StripeConnectAccount{} = connect_account
-           }
-         } = project <-
-           get_project(project_id),
-         %User{
-           stripe_platform_card: %StripePlatformCard{} = platform_card,
-           stripe_platform_customer: %StripePlatformCustomer{} = platform_customer
-         } = user <-
-           get_user(user_id),
-         {:ok, %StripeConnectSubscription{} = stripe_connect_subscription} <-
-           do_find_or_create(attributes, connect_account, plan, project, platform_card, platform_customer, quantity, user),
-         _project <-
-           ProjectService.update_project_totals(project)
+  def find_or_create(%{"project_id" => project_id, "quantity" => _, "user_id" => user_id} = attributes) do
+    with {:ok, %Project{} = project} <- get_project(project_id) |> ProjectSubscribable.validate,
+         {:ok, %User{} = user} <- get_user(user_id) |> UserCanSubscribe.validate
     do
-      {:ok, stripe_connect_subscription}
+      {:ok, %StripeConnectSubscription{} = subscription} = do_find_or_create(project, user, attributes)
+
+      ProjectService.update_project_totals(project)
+      DonationGoalsService.update_project_goals(project)
+
+      {:ok, subscription}
     else
+      {:error, :project_not_ready} -> {:error, :project_not_ready}
+      {:error, :user_not_ready} -> {:error, :user_not_ready}
```

Comment:
I'd like to keep it explicit because I'm expecting we might be doing custom handling/errors at some point. I can move that into a comment above the line, though.